app/controllers/competitors_controllers.rb

```diff
@@ -0,0 +1,9 @@
+MyApp.get "/competitors" do
+  x = Competitor.new
+  x.first_name = params["fname"]
+  x.last_name = params["lname"]
+  x.school_id = params["school"]
+  x.save
+  @y = [x.first_name, x.last_name, x.school_id]
```

Comment:
Interesting solution. You could also make `x` into `@x` and refer to its getter methods in the view directly.